pages/playground.js

```diff
@@ -1,140 +1,271 @@
-            setFormatComplete(false);
-            setValidateComplete(false);
-            setSuccessMsg("");
-            setErrMsg("");
-          }}
-        />
-        <div className="flex flex-row justify-end mb-3 gap-2">
-          {!formatComplete && (
-            <Button text="Format" onClick={handleFormatJson} primary={false} />
-          )}
-          {formatComplete && !validateComplete && (
-            <Button
-              text="Validate"
-              onClick={handleValidateJson}
-              primary={false}
-            />
-          )}
-          {formatComplete && validateComplete && (
-            <Button text="Preview" onClick={handlePreview} primary={true} />
-          )}
-        </div>
-
-        {previewModalData && previewModalState && (
-          <PreviewModal
-            toggle={() => setPreviewModalState(!previewModalState)}
-            data={previewModalData}
-          />
-        )}
-      </Page>
-    </>
-  );
+ const [profileJson, setProfileJson] = useState("");
+ const [eventJson, setEventJson] = useState("");
+ const [validateComplete, setValidateComplete] = useState(false);
+ const [validateEventComplete, setValidateEventComplete] = useState(false);
+ const [formatComplete, setFormatComplete] = useState(false);
+ const [formatEventComplete, setFormatEventComplete] = useState(false);
+ const [errorMessage, setErrMsg] = useState("");
+ const [errorEventMessage, setEventErrMsg] = useState("");
+ const [successMessage, setSuccessMsg] = useState("");
+ const [successEventMessage, setSuccessEventMsg] = useState("");
+ const [gitUsername, setGitUsername] = useState("");
+ const [previewModalState, setPreviewModalState] = useState(false);
+ const [previewEventModalState, setPreviewEventModalState] = useState(false);
+ const [previewModalData, setPreviewModalData] = useState();
+ const [previewEventModalData, setEventPreviewModalData] = useState();
+
+
+
+
+ const handleValidateJson = () => {
+   try {
+     JSON.parse(profileJson);
+     setSuccessMsg("Valid Json");
+     setErrMsg("");
+     setValidateComplete(true);
+     return true;
+   } catch (err) {
+     setErrMsg(err.toString());
+     setError(true);
+     setSuccessMsg("");
+     return false;
+   }
+ };
+
+
+ const handleValidateEventJson = () => {
+   try {
+     JSON.parse(eventJson);
+     setSuccessEventMsg("Valid Json");
+     setEventErrMsg("");
+     setValidateEventComplete(true);
+     return true;
+   } catch (err) {
+     setEventErrMsg(err.toString());
+     setError(true);
+     setSuccessEventMsg("");
+     return false;
+   }
+ };
+
+
+ const handleFormatJson = () => {
+   try {
+     let parsedJson = JSON.parse(profileJson);
+     let formattedJson = JSON.stringify(parsedJson, undefined, 4);
+     setProfileJson(formattedJson);
+     setErrMsg("");
+     setFormatComplete(true);
+   } catch (err) {
+     setErrMsg(err.toString());
+     setSuccessMsg("");
+   }
+ };
+
+
+ const handleFormatEventJson = () => {

```

Comment:
Can be combined with `handleFormatJson` as looks like the same code

Reply:
We have tried to combine these by passing in parameters, but it results in other errors.  Any recommendations on how to cleanly combine them?  Otherwise, do you think they could be left as is as they do have separate functionality?